pilot/pkg/kube/inject/app_probe.go

```diff
@@ -80,45 +101,68 @@ func extractStatusPort(sidecar *corev1.Container) int {
 	return -1
 }
 
-// rewriteProbe changes application containers' probe to point to sidecar's status port.
-func rewriteProbe(probe *corev1.Probe, appProbers *status.KubeAppProbers,
-	newURL string, statusPort int, portMap map[string]int32) {
+// convertAppProber returns a overwritten `HTTPGetAction` for pilot agent to take over.
+func convertAppProber(probe *corev1.Probe, newURL string, statusPort int) *corev1.HTTPGetAction {
 	if probe == nil || probe.HTTPGet == nil {
-		return
+		return nil
 	}
-	httpGet := probe.HTTPGet
-
-	// Save app probe config to pass to pilot agent later.
-	savedProbe := proto.Clone(probe.HTTPGet).(*corev1.HTTPGetAction)
-	(*appProbers)[newURL] = savedProbe
-	// A named port, resolve by looking at port map.
-	if httpGet.Port.Type == intstr.String {
-		port, exists := portMap[httpGet.Port.StrVal]
-		if !exists {
-			log.Errorf("named port not found in the map skip rewriting probing %v", *probe)
-			return
-		}
-		savedProbe.Port = intstr.FromInt(int(port))
-	}
-	// Change the application csince ontainer prober config.
-	httpGet.Port = intstr.FromInt(statusPort)
-	httpGet.Path = newURL
+	c := proto.Clone(probe.HTTPGet).(*corev1.HTTPGetAction)
```

Comment:
Why we need this Clone？Isn't it ok to simply construct `HTTPGetAction`?

Reply:
`HTTPGetAction` has many fields other than port and path. We want to preserve all of it when passing to pilot agent, e.g. http header.

Reply:
How about directly modify probe.HTTPGet, and return it? I doubt `proto.Clone` is not a high performance ops.

If really need deep copy, should use `HTTPGetAction.DeepCopy` instead.

Reply:
Done.